client/blocks/import/capture/capture-input.tsx

```diff
@@ -69,13 +72,124 @@ const CaptureInput: FunctionComponent< Props > = ( props ) => {
 		}
 	}
 
+	function isIDN( url: string ) {
+		try {
+			// Regex to extract the hostname from the URL.
+			const urlRegex = /^(?:https?:\/\/)?(?:www\.)?([^/]+)/i;
+			const match = url.match( urlRegex );
+
+			if ( ! match ) {
+				return false;
+			}
+
+			// Extract the hostname.
+			const hostname = match[ 1 ];
+
+			// Check for non-ASCII characters
+			for ( let i = 0; i < hostname.length; i++ ) {
+				if ( hostname.charCodeAt( i ) > 127 ) {
+					return true;
+				}
+			}
+
+			// Check if the hostname starts with the Punycode prefix.
+			if ( hostname.startsWith( 'xn--' ) ) {
+				return true;
+			}
+		} catch ( e ) {
+			return false;
+		}
+	}
+
 	function validateUrl( url: string ) {
 		const isValid = CAPTURE_URL_RGX.test( url );
 		setIsValid( isValid );
+		const tempValidationMessage = isValid ? '' : getValidationMessage( url );
+		setValidationMessage( tempValidationMessage );
+	}
+
+	// TODO: Just return the translated string directly from getValidationMessage once we have translations for all messages.
+	const errorMessages = {
```

Comment:
Suggestion: Move all error message logic to a private component, since is this pretty big.

Reply:
Thank you, done here https://github.com/Automattic/wp-calypso/pull/91748/commits/228d668523ae30f2e7df7fbf5e4a90c0518c4348